onnxruntime/test/unittest_main/test_main.cc

```diff
@@ -93,6 +93,8 @@ int TEST_MAIN(int argc, char** argv) {
 
   ORT_TRY {
     ortenv_setup();
+    // TODO: Fix the C API issue
+    atexit(ortenv_teardown); // If we don't do this, it will crash
```

Comment:
Atexit functions get executed after the main function is returned, especially , it is after the "::google::protobuf::ShutdownProtobufLibrary" , which means in the teardown function we cannot call all protobuf functions, which is not possible.  Also, it will also change the destruction order between function local statics and this "ortenv_teardown" function. Because basically the function local statics will use atexit function to register their destructors.  Therefore, here the ordering is really very tricky. I would suggest keeping it unchanged unless there is an urgent need.

Reply:
would using `gsl::finally` instead of `atexit` work?

e.g., `#include <gsl/gsl>`, then,

```suggestion
    auto clean_up_ortenv = gsl::finally(ortenv_teardown); // If we don't do this, it will crash
```

Reply:
@edgchen1 tried with `gsl::finally`, no luck. The problem is abseil parsing is just calling for exit() if --help flag is used, so only `atexit` has effect in this way.
@snnn I see your point, I agree this ordering is indeed tricky, and unfortunately abseil and gtest don't leave enough control on possible hooks on help exit.

I've pushed one last tentative. This time, `ortenv_setup` is called only if --help is not specified. This should not alter the ordering for test execution and --help is not crashing, as ortenv_setup is not called in that case